turbine/src/cluster_nodes.rs

```diff
@@ -310,6 +311,11 @@ pub fn new_cluster_nodes<T: 'static>(
     if broadcast {
         weighted_shuffle.remove_index(index[&self_pubkey]);
     }
+    // Paranoid check to ensure the correct operation of the weighted shuffle.
+    assert!(
```

Comment:
maybe `debug_assert` ?

Reply:
This does not get called very often (once every few seconds), and this ensures the thing crashes early (rather than in any of the unwraps in the code)